cheddar/cheddar-integration-aws/src/main/java/com/clicktravel/infrastructure/persistence/aws/dynamodb/QuerySpecBuilder.java

```diff
@@ -0,0 +1,84 @@
+import com.clicktravel.cheddar.infrastructure.persistence.database.query.CompoundAttributeQuery;
+import com.clicktravel.cheddar.infrastructure.persistence.database.query.Operators;
+import com.clicktravel.cheddar.infrastructure.persistence.exception.PersistenceResourceFailureException;
+
+public class QuerySpecBuilder {
+
+    public static <T extends Item> QuerySpec build(final AttributeQuery attributeQuery, final Class<T> itemClass) {
+        if (CompoundAttributeQuery.class.isAssignableFrom(attributeQuery.getClass())) {
+            final CompoundAttributeQuery compoundAttributeQuery = (CompoundAttributeQuery) attributeQuery;
+            return buildWithHashAndRangeKey(compoundAttributeQuery, itemClass);
+        }
+
+        return buildWithHashKey(attributeQuery);
+    }
+
+    private static QuerySpec buildWithHashKey(final AttributeQuery attributeQuery) {
+        return new QuerySpec().withHashKey(attributeQuery.getAttributeName(),
+                attributeQuery.getCondition().getValues().iterator().next());
+    }
+
+    private static <T extends Item> QuerySpec buildWithHashAndRangeKey(
+            final CompoundAttributeQuery compoundAttributeQuery, final Class<T> itemClass) {
+        final Set<String> supportingConditionValues = compoundAttributeQuery.getSupportingCondition().getValues();
+        validateSupportingConditionValues(supportingConditionValues);
+
+        final QuerySpec querySpec = buildWithHashKey(compoundAttributeQuery);
+        addRangeKeyConditionToQuerySpec(querySpec, compoundAttributeQuery, itemClass);
+
+        return querySpec;
+    }
+
+    private static <T extends Item> void addRangeKeyConditionToQuerySpec(final QuerySpec querySpec,
+            final CompoundAttributeQuery compoundAttributeQuery, final Class<T> itemClass) {
+        final String supportingConditionValue = compoundAttributeQuery.getSupportingCondition().getValues().iterator()
+                .next();
+        final Operators comparisonOperator = compoundAttributeQuery.getSupportingCondition().getComparisonOperator();
+        final Class<?> clazz = compoundAttributeQuery.getSupportingAttributeType(itemClass);
+
+        try {
+            final Object value = clazz.getConstructor(String.class).newInstance(supportingConditionValue);
+
+            final RangeKeyCondition rangeKeyCondition = RangeKeyConditionBuilder
+                    .build(compoundAttributeQuery.getSupportingAttributeName(), value, comparisonOperator);
+            querySpec.withRangeKeyCondition(rangeKeyCondition);
+        } catch (final Exception e) {
+            throw new PersistenceResourceFailureException(
+                    String.format("Could not add range key condition for query: %s on item %s.", compoundAttributeQuery,
+                            itemClass.getSimpleName()),
+                    e);
+        }
+    }
+
+    private static void validateSupportingConditionValues(final Set<String> supportingConditionValues) {
```

Comment:
`supportingConditionValues` is fine here (as opposed to `supportingConditionStringValues`), as there is no competing variable with the different type